pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -509,13 +518,23 @@ func (l *AccessCredentialManager) watchSecrets(vmi *v1.VirtualMachineInstance) {
 				reload = true
 				reportedErr = true
 				logger.Reason(err).Errorf("Error encountered setting password for user [%s]", user)
-
-				l.reportAccessCredentialResult(false, fmt.Sprintf("Error encountered setting password for user [%s]: %v", user, err))
+				l.reportAccessCredentialResult(api.AccessCredentialMetadata{
+					Succeeded: false,
+					Message:   fmt.Sprintf("Error encountered setting password for user [%s]: %v", user, err),
+				})
 				continue
 			}
 		}
 		if !reportedErr {
-			l.reportAccessCredentialResult(true, "")
+			acMetadata := api.AccessCredentialMetadata{
+				Succeeded: true,
+			}
+			if usedDeprecatedSshFlow {
+				acMetadata.DeprecatedMethod = true
+				acMetadata.Message = "Used deprecated method to set SSH keys. It will be removed in a future release. Update qemu guest agent to 5.2 or newer to use the new method."
```

Comment:
Any chance to point out the user which will be the new method here?

Reply:
I've changed the sentence to not mention a new method. I think we don't want to give user unnecessary technical details.